src/main/java/fi/aalto/cs/apluscourses/ui/base/TreeView.java

```diff
@@ -130,4 +157,17 @@ public void mouseClicked(@NotNull MouseEvent e) {
       }
     }
   }
+
+  private static class MyTreePathCoder extends TreePathCoder<String> {
```

Comment:
..... `StringTreePathCoder`?

Reply:
Or even better, `TreePathString(En)coder`

Reply:
Hmm, maybe, but... it's not a generic to-string-encoder but also aware of the semantics of the node objects. It's a `TreePathEncoder` used by that class, `TreeView`'s own `TreePathEncoder` (hence, "My").

Reply:
Well, yes it's not generic, but it's also private so no one is going to use it by mistake, I do consider this "My" and "Our" prefixing quite indescriptive, and it's quite similar to the "Impl" suffix naming perversion